openshift-knative-operator/pkg/common/cabundle_test.go

```diff
@@ -0,0 +1,108 @@
+package common
+
+import (
+	"path/filepath"
+	"testing"
+
+	"k8s.io/client-go/kubernetes/scheme"
+
+	util "knative.dev/operator/pkg/reconciler/common/testing"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+func TestApplyCABundlesTransform(t *testing.T) {
+	for _, tt := range []struct {
+		name     string
+		actual   corev1.PodSpec
+		expected corev1.PodSpec
+	}{{
+		name:   "SSL_cert_injectior",
+		actual: *podSpecable(t),
+		expected: *podSpecable(t,
+			withEnvs(
+				corev1.EnvVar{
+					Name:  "SSL_CERT_DIR",
+					Value: "/ocp-serverless-custom-certs:/etc/pki/tls/certs",
+				},
+			),
+			withVolumes(corev1.Volume{
+				Name: TrustedCAConfigMapVolume,
+				VolumeSource: corev1.VolumeSource{
+					ConfigMap: &corev1.ConfigMapVolumeSource{
+						LocalObjectReference: corev1.LocalObjectReference{Name: TrustedCAConfigMapName},
+						Items: []corev1.KeyToPath{
+							{
+								Key:  TrustedCAKey,
+								Path: TrustedCAKey,
+							},
+						},
+					},
+				},
+			}),
+			withVolumeMounts(
+				corev1.VolumeMount{
+					Name:      TrustedCAConfigMapVolume,
+					MountPath: filepath.Join("/ocp-serverless-custom-certs", TrustedCAKey),
+					ReadOnly:  true,
+				}),
+		),
+	}} {
+		t.Run(tt.name, func(t *testing.T) {
+
+			caBundleTransform := ApplyCABundlesTransform()
+
+			// test for deployment
+			unstructuredDeployment := util.MakeUnstructured(t, util.MakeDeployment(tt.name, tt.actual))
+			caBundleTransform(&unstructuredDeployment)
+			deployment := &appsv1.Deployment{}
+			err := scheme.Scheme.Convert(&unstructuredDeployment, deployment, nil)
+			util.AssertEqual(t, err, nil)
+			util.AssertDeepEqual(t, deployment.Spec.Template.Spec, tt.expected)
+
+			// TODO: test for statefulset. Will come in next 1.11 upstream z-release...
```

Comment:
will come soon... `util.MakeDeployment` was recently introduced on upstream 1.11 (and later). We will update to that soon-ish